### PR TITLE
ANDROID: Update to compile with modern NDKs

### PR DESCRIPTION
**Pull request by ccawley2011**
This is partially based on scummvm/scummvm#1278 with some additional ResidualVM-specific changes.

### CODE REVIEW

**Comment by Botje** [comment]
@klusark are you up for updating the Android NDK/SDK used on your Jenkins setup? Or should we move to a proper CI system?

**Comment by klusark** [comment]
@Botje At this point it's probably better to move to a proper CI system, since what you get for free these days is quite good, and it's easier for anyone to make changes to it if it's in the git repo.

**Comment by Botje** [comment]
@klusark That was my idea too. Can you upload your signing keypair to the Google Play console so we can sign automatically built releases then?

File: backends/platform/android/android.mk [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # These must be incremented for each market upload
 ANDROID_VERSIONCODE = 1000
 
-ANDROID_TARGET_VERSION = 15
+ANDROID_TARGET_VERSION = 23
```
Comment (Botje) [inline backends/platform/android/android.mk]: We'll need to bump this in the future to actually get the thing published, as google [requires version 26](https://developer.android.com/distribute/best-practices/develop/target-sdk) for updates, and 29 from Nov 2019 on. But let's get all of this merged first :)

**Comment by klusark** [comment]
@Botje Done. Didn't realize there was an option to upload the key before. Makes it much easier to manage.

**Pull request merged**